k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/GenericK8sResourceHandler.kt

```diff
@@ -228,4 +233,128 @@ abstract class GenericK8sResourceHandler <S: GenericK8sLocatable, R: K8sManifest
         }
         return manifest
     }
+
+    @Suppress("UNCHECKED_CAST")
+    suspend fun getFluxK8sResources(resource: Resource<S>, artifact: DeliveryArtifact, correlationId: String, environment: String?): R? {
+        val namespace = getNamespace(resource)
+        val repoNameInClouddriver: String
+        val repoNamespace: String
+        when (artifact) {
+            is GitRepoArtifact -> {
+                repoNamespace = artifact.namespace
+                repoNameInClouddriver = environment?.let {
+                    "${artifact.kind} ${artifact.name}-$it"
+                } ?: run{"${artifact.kind} ${artifact.name}"}
+            }
+            // TODO implement other types
+            else -> throw InvalidArtifact("artifact is not a supported artifact type. artifact: $artifact")
+        }
+        // need to wrap async with coroutineScope to catch exceptions correctly
+        try {
+            return coroutineScope {
+                val repoJob = async {
+                    cloudDriverK8sService.getK8sResource(
+                        resource.serviceAccount,
+                        resource.spec.locations.account,
+                        repoNamespace,
+                        repoNameInClouddriver
+                    )
+                }
+                val resourceJob = async {
+                    cloudDriverK8sService.getK8sResource(
+                        resource.serviceAccount,
+                        resource.spec.locations.account,
+                        namespace,
+                        resource.spec.template!!.kindQualifiedName()
+                    )
+                }
```

Comment:
can we defer to existing `getK8sResource` function? the fewer of these, the easier to later on modify or change queries for resources etc, if needed.